jvm/src/test/java/com/microsoft/bond/protocol/BinaryStreamReaderTest.java

```diff
@@ -0,0 +1,693 @@
+                (byte) 0xDB, (byte) 0x0F, (byte) 0x49, (byte) 0x40,
+
+                // -Infinity, bit pattern:  0xFF800000
+                (byte) 0x00, (byte) 0x00, (byte) 0x80, (byte) 0xFF,
+
+                // +Infinity, bit pattern:  0x7F800000
+                (byte) 0x00, (byte) 0x00, (byte) 0x80, (byte) 0x7F,
+
+                // A NaN (canonical for some mainstream JVM implementations), bit pattern:  0x7FC00000
+                // (since "canonical Java NaN" is unspecified, this is treated as any other NaN representation)
+                (byte) 0x00, (byte) 0x00, (byte) 0xC0, (byte) 0x7F,
+
+                // A NaN (non-canonical), bit pattern:  0xFFC00000
+                (byte) 0x00, (byte) 0x00, (byte) 0xC0, (byte) 0xFF,
+
+                // A NaN (non-canonical), bit pattern:  0x7F800001
+                (byte) 0x01, (byte) 0x00, (byte) 0x80, (byte) 0x7F
+        };
+        ByteArrayInputStream bais = new ByteArrayInputStream(inputBytes);
+        BinaryStreamReader r = new BinaryStreamReader(bais);
+
+        // read first (0)
+        try {
+            float v = r.readFloat();
+            assertEquals(0F, v, 0F);
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0x00000000, vb);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read second (PI)
+        try {
+            float v = r.readFloat();
+            assertEquals((float) Math.PI, v, 0F);
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0x40490FDB, vb);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read third (-Infinity)
+        try {
+            float v = r.readFloat();
+            assertTrue(v == Float.NEGATIVE_INFINITY);
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0xFF800000, vb);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read fourth (+Infinity)
+        try {
+            float v = r.readFloat();
+            assertTrue(v == Float.POSITIVE_INFINITY);
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0x7F800000, vb);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read fifth (NaN, canonical)
+        try {
+            float v = r.readFloat();
+            assertTrue(Float.isNaN(v));
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0x7FC00000, vb);
+        } catch (IOException e) {
+            fail("IOException can't be thrown here: " + e);
+        }
+
+        // read sixth (NaN, non-canonical)
+        try {
+            float v = r.readFloat();
+            assertTrue(Float.isNaN(v));
+            int vb = Float.floatToRawIntBits(v);
+            assertEquals(0xFFC00000, vb);
```

Comment:
Agree, the isNaN test is redundant. But doesn't hurt.

---
In reply to: [119250270](https://github.com/Microsoft/bond/pull/465#discussion_r119250270) [](ancestors = 119250270)